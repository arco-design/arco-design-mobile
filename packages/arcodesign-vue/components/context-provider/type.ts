export interface ContextInstallOptions {
    /**
     * 组件名称前缀
     * @en Component prefix
     * @default "Arco"
     */
    componentPrefix?: string;
}

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
}
