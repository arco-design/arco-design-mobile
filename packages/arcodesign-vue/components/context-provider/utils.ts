import { InjectionKey, inject } from 'vue';
import { ContextInstallOptions, GlobalContextParams } from './type';

export const defaultInstallOptions: Required<ContextInstallOptions> = {
    componentPrefix: 'Arco',
};

export const defaultContext: Required<GlobalContextParams> = {
    prefixCls: 'arco',
};

export const configProviderInjectionKey: InjectionKey<GlobalContextParams> =
    Symbol('ArcoContextProvider');

export function getComponentPrefix(options?: ContextInstallOptions) {
    return options?.componentPrefix || defaultInstallOptions.componentPrefix;
}

export function getPrefixCls(componentName?: string) {
    const contextProvider = inject(configProviderInjectionKey, defaultContext);
    const prefix = contextProvider.prefixCls;
    if (componentName) {
        return `${prefix}-${componentName}`;
    }
    return prefix;
}
