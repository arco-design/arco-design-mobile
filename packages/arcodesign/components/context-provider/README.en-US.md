### Others

# ContextProvider 

Global data control component, used to replace global data.

======

> Props

|Property|Description|Type|DefaultValue|
|----------|-------------|------|------|
|prefixCls|Component classname prefix|string|"arco"|
|system|Manually control the current system, and the incoming value will be used directly after being passed in\. It is applicable when the initial value of the system needs to be specified in the ssr scenario\.|"" \| "pc" \| "android" \| "ios"|""|
|useDarkMode|Whether to use dark mode|boolean|false|
|isDarkMode|Whether it is in dark mode|boolean|false|
|theme|Theme variable\. The css variable will be replaced online after input\. The less variable needs to be set|Record\<string, string\>|-|
|locale|Internationalized language configuration|ILocale|-|
|useRtl|Whether to use rtl|boolean|false|

> ILocale

|Property|Description|Type|
|----------|-------------|------|
|locale|Language Type|string|
|LoadMore|-|\{ loadMoreText: string; loadingText: string; prepareText: string; noDataText: string; failLoadText: string; prepareScrollText: string; prepareClickText: string; \}|
|Picker|-|\{ okText: string; cancelText: string; \}|
|Tag|-|\{ addTag: string; \}|
|Dialog|-|\{ okText: string; cancelText: string; \}|
|SwipeLoad|-|\{ normalText: string; activeText: string; \}|
|PullRefresh|-|\{ loadingText: string; pullingText: string; finishText: string; loosingText: string; \}|
|DropdownMenu|-|\{ select: string; \}|
|Pagination|-|\{ previousPage: string; nextPage: string; \}|
|Image|-|\{ loadError: string; \}|
|ImagePicker|-|\{ loadError: string; \}|
|SearchBar|-|\{ placeholder: string; cancelBtn: string; \}|
|Stepper|-|\{ minusButtonName: string; addButtonName: string; \}|
|Keyboard|-|\{ confirm: string; \}|
|Form|-|\{ required: string; type: \{ email: string; url: string; string: string; number: string; array: string; object: string; boolean: string; \}; number: \{ min: string; max: string; equal: string; range: string; positive: string; negative: string; \}; string: \{ \.\.\.; \}; array: \{ \.\.\.; \}; object: \{ \.\.\.; \}; boolean: \{ \.\.\.; \}; \}|
