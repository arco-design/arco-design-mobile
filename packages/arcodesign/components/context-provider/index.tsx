import React, { createContext, useState, useEffect, useCallback } from 'react';
import { addCssRules, removeCssStyleDom, defaultLocale, ILocale } from '@arco-design/mobile-utils';

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
    /**
     * 手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用
     * @en Manually control the current system, and the incoming value will be used directly after being passed in. It is applicable when the initial value of the system needs to be specified in the ssr scenario.
     * @default ""
     */
    system?: 'pc' | 'android' | 'ios' | '';
    /**
     * 是否使用暗黑模式
     * @en Whether to use dark mode
     * @default false
     */
    useDarkMode?: boolean;
    /**
     * 是否处于暗黑模式，指定后以指定的值为准
     * @en Whether it is in dark mode
     * @default false
     */
    isDarkMode?: boolean;
    /**
     * 主题变量，传入后将在线替换css变量，需设置less变量 @use-css-vars: 1
     * @en Theme variable. The css variable will be replaced online after input. The less variable needs to be set @use-css-vars: 1
     */
    theme?: Record<string, string>;
    /**
     * 国际化语言包配置
     * @en Internationalized language configuration
     */
    locale?: ILocale;
    /**
     * 是否使用Rtl模式
     * @en Whether to use rtl
     * @default false
     */
    useRtl?: boolean;
}

export const defaultContext: GlobalContextParams = {
    prefixCls: 'arco',
    system: '',
    useDarkMode: false,
    isDarkMode: false,
    locale: defaultLocale,
    useRtl: false,
};

export const GlobalContext = createContext<GlobalContextParams>(defaultContext);

export interface ContextProviderProps extends GlobalContextParams {
    children: React.ReactNode;
}

export type WithGlobalContext<T> = T & { context?: GlobalContextParams };

/**
 * 全局数据控制组件，用于替换全局数据。
 * @en Global data control component, used to replace global data.
 * @type 其他
 * @type_en Others
 * @name 全局配置
 * @name_en ContextProvider
 */
export default function ContextProvider(props: ContextProviderProps) {
    const {
        children,
        useDarkMode,
        isDarkMode,
        theme,
        locale = defaultLocale,
        ...restProps
    } = props;
    const [isDarkModeState, setIsDarkModeState] = useState<boolean>();
    const changeDarkMode = useCallback(
        (res: MediaQueryListEvent) => setIsDarkModeState(res.matches),
        [],
    );

    useEffect(() => {
        const matchMedia = window.matchMedia('(prefers-color-scheme: dark)');
        if (useDarkMode) {
            const dark = matchMedia.matches;
            setIsDarkModeState(dark);
            if (typeof matchMedia.addEventListener === 'function') {
                matchMedia.addEventListener('change', changeDarkMode);
            } else if (typeof matchMedia.addListener === 'function') {
                matchMedia.addListener(changeDarkMode);
            }
        }
        return () => {
            if (useDarkMode) {
                if (typeof matchMedia.removeEventListener === 'function') {
                    matchMedia.removeEventListener('change', changeDarkMode);
                } else if (typeof matchMedia.removeListener === 'function') {
                    matchMedia.removeListener(changeDarkMode);
                }
            }
        };
    }, [useDarkMode]);

    useEffect(() => {
        if (!theme) {
            return;
        }
        addCssRules('arcoCustomTheme', theme);
        return () => {
            removeCssStyleDom('arcoCustomTheme');
        };
    }, [theme]);

    return (
        <GlobalContext.Provider
            value={{
                ...defaultContext,
                ...restProps,
                isDarkMode: isDarkMode === void 0 ? isDarkModeState : isDarkMode,
                useDarkMode,
                locale: locale || defaultLocale,
            }}
        >
            {children}
        </GlobalContext.Provider>
    );
}

export const ContextLayout = GlobalContext.Consumer;

export function CompWithGlobalContext<P extends JSX.IntrinsicAttributes>(
    Component: React.FunctionComponent<P>,
) {
    return function (props: WithGlobalContext<P>) {
        const { context: propsContext, ...others } = props;
        return (
            <ContextLayout>
                {context => (
                    <ContextProvider {...{ ...context, ...propsContext }}>
                        <Component {...(others as P)} />
                    </ContextProvider>
                )}
            </ContextLayout>
        );
    };
}
