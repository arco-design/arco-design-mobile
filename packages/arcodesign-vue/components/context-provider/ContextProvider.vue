<script lang="ts" setup>
import { provide } from 'vue';
import { GlobalContextParams } from './type';
import { configProviderInjectionKey, defaultContext } from './utils';

const props = withDefaults(defineProps<GlobalContextParams>(), defaultContext);
provide(configProviderInjectionKey, props);
</script>
<template>
    <slot />
</template>
