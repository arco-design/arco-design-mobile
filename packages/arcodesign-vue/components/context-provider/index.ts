import ContextProvider from './ContextProvider.vue';
import { installFactory } from '../_helpers/install';

export type * from './type';
export * from './utils';

const install = installFactory(ContextProvider);

export { ContextProvider };

export default {
    install,
};
