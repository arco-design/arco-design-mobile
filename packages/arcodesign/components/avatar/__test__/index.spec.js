import React from 'react';
import { mount } from 'enzyme';
import demoTest from '../../../tests/demoTest';
import mountTest from '../../../tests/mountTest';
import Avatar from '..';
import { defaultContext } from '../../context-provider';

const demoAvatarSrc = 'https://sf1-cdn-tos.toutiaostatic.com/obj/arco-mobile/_static_/small_image_5.jpg';
const prefix = `${defaultContext.prefixCls}-avatar`;

demoTest('avatar');

mountTest(Avatar, 'Avatar', {
    src: demoAvatarSrc,
    size: 'large',
});

describe('Avatar', () => {

    it('src renders correctly', () => {
        const wrapper = mount(
        <Avatar
            src={demoAvatarSrc}/>,
        );
        expect(wrapper.find('.image-content').length).toBe(1);
    });

    it('group renders correctly', () => {
        const wrapper = mount(
            <Avatar.Group size="small">
                <Avatar textAvatar="尼采尼采尼采尼采尼采" avatarStyle={{backgroundColor: '#7BC616'}} />
                <Avatar textAvatar="M" avatarStyle={{backgroundColor: '#14C9C9'}} />
                <Avatar textAvatar="X" avatarStyle={{backgroundColor: '#168CFF'}} />
                <Avatar textAvatar="Z" avatarStyle={{backgroundColor: '#FF7D00'}} />
                <Avatar textAvatar="JD" avatarStyle={{backgroundColor: '#FFC72E'}} />
                <Avatar textAvatar="JD" size='medium' avatarStyle={{backgroundColor: '#FFC72E'}} />
            </Avatar.Group>,
        );
        expect(wrapper.find(`.${prefix}-wrapper`).length).toBe(6);
    });
})
