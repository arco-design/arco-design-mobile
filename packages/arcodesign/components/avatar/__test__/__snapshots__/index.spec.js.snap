// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`avatar demo test avatar demo: auto-font-size.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="avatar-demo-box"
  >
    <div
      class="arco-avatar-wrapper small circle arco-avatar-wrapper-shape-circle"
    >
      <div
        class="arco-avatar arco-avatar-size-small small arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
      >
        <span
          class="arco-avatar-text arco-avatar-text-size-small"
          style="transform: scale(1);"
        >
          Arco
        </span>
      </div>
    </div>
    <div
      class="arco-avatar-wrapper avatar-left-margin small circle arco-avatar-wrapper-shape-circle"
    >
      <div
        class="arco-avatar arco-avatar-size-small small arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
        style="background-color: rgb(255, 125, 0);"
      >
        <span
          class="arco-avatar-text arco-avatar-text-size-small"
          style="transform: scale(1);"
        >
          ByteDancer
        </span>
      </div>
    </div>
    <div
      class="arco-avatar-wrapper avatar-left-margin small square arco-avatar-wrapper-shape-square"
    >
      <div
        class="arco-avatar arco-avatar-size-small small arco-avatar-shape-square square arco-text-avatar arco-avatar-mode-text"
        style="background-color: rgb(201, 205, 212);"
      >
        <span
          class="arco-avatar-text arco-avatar-text-size-small"
          style="transform: scale(1);"
        >
          Garth
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`avatar demo test avatar demo: avatar-with-info.md renders correctly 1`] = `
<DocumentFragment>
  <div
    style="margin: -16px 0px;"
  >
    <div
      class="arco-avatar-wrapper large circle arco-avatar-wrapper-shape-circle arco-avatar-wrapper-with-info arco-avatar-wrapper-with-info-size-large with-info"
    >
      <div
        class="arco-avatar arco-avatar-size-large large arco-avatar-shape-circle circle arco-image-avatar arco-avatar-mode-image"
      >
        <div
          class="arco-image all-border-box pc arco-avatar-img"
        >
          <div
            class="image-content image-bottom-overlap"
          >
            <div
              class="image-placeholder"
            />
          </div>
          <div
            class="image-container loading animate"
            style="transition-duration: 200ms;"
          />
        </div>
      </div>
      <div
        class="arco-avatar-info"
      >
        <div
          class="arco-avatar-name arco-avatar-name-size-large"
        >
          Username
        </div>
        <div
          class="arco-avatar-desc arco-avatar-desc-size-large"
        >
          Auxiliary information
        </div>
      </div>
    </div>
    <div
      class="avatar-divider"
    />
    <div
      class="arco-avatar-wrapper medium circle arco-avatar-wrapper-shape-circle arco-avatar-wrapper-with-info arco-avatar-wrapper-with-info-size-medium with-info"
    >
      <div
        class="arco-avatar arco-avatar-size-medium medium arco-avatar-shape-circle circle arco-image-avatar arco-avatar-mode-image"
      >
        <div
          class="arco-image all-border-box pc arco-avatar-img"
        >
          <div
            class="image-content image-bottom-overlap"
          >
            <div
              class="image-placeholder"
            />
          </div>
          <div
            class="image-container loading animate"
            style="transition-duration: 200ms;"
          />
        </div>
      </div>
      <div
        class="arco-avatar-info"
      >
        <div
          class="arco-avatar-name arco-avatar-name-size-medium"
        >
          Username
        </div>
        <div
          class="arco-avatar-desc arco-avatar-desc-size-medium"
        >
          Auxiliary information
        </div>
      </div>
    </div>
    <div
      class="avatar-divider"
    />
    <div
      class="arco-avatar-wrapper small circle arco-avatar-wrapper-shape-circle arco-avatar-wrapper-with-info arco-avatar-wrapper-with-info-size-small with-info"
    >
      <div
        class="arco-avatar arco-avatar-size-small small arco-avatar-shape-circle circle arco-image-avatar arco-avatar-mode-image"
      >
        <div
          class="arco-image all-border-box pc arco-avatar-img"
        >
          <div
            class="image-content image-bottom-overlap"
          >
            <div
              class="image-placeholder"
            />
          </div>
          <div
            class="image-container loading animate"
            style="transition-duration: 200ms;"
          />
        </div>
      </div>
      <div
        class="arco-avatar-info"
      >
        <div
          class="arco-avatar-name arco-avatar-name-size-small"
        >
          Username
        </div>
        <div
          class="arco-avatar-desc arco-avatar-desc-size-small"
        >
          Auxiliary information
        </div>
      </div>
    </div>
    <div
      class="avatar-divider"
    />
    <div
      class="arco-avatar-wrapper smaller circle arco-avatar-wrapper-shape-circle arco-avatar-wrapper-with-info arco-avatar-wrapper-with-info-size-smaller with-info"
    >
      <div
        class="arco-avatar arco-avatar-size-smaller smaller arco-avatar-shape-circle circle arco-image-avatar arco-avatar-mode-image"
      >
        <div
          class="arco-image all-border-box pc arco-avatar-img"
        >
          <div
            class="image-content image-bottom-overlap"
          >
            <div
              class="image-placeholder"
            />
          </div>
          <div
            class="image-container loading animate"
            style="transition-duration: 200ms;"
          />
        </div>
      </div>
      <div
        class="arco-avatar-info"
      >
        <div
          class="arco-avatar-name arco-avatar-name-size-smaller"
        >
          Username
        </div>
        <div
          class="arco-avatar-desc arco-avatar-desc-size-smaller"
        >
          Auxiliary information
        </div>
      </div>
    </div>
    <div
      class="avatar-divider"
    />
    <div
      class="arco-avatar-wrapper ultra-small circle arco-avatar-wrapper-shape-circle arco-avatar-wrapper-with-info arco-avatar-wrapper-with-info-size-ultra-small with-info"
    >
      <div
        class="arco-avatar arco-avatar-size-ultra-small ultra-small arco-avatar-shape-circle circle arco-image-avatar arco-avatar-mode-image"
      >
        <div
          class="arco-image all-border-box pc arco-avatar-img"
        >
          <div
            class="image-content image-bottom-overlap"
          >
            <div
              class="image-placeholder"
            />
          </div>
          <div
            class="image-container loading animate"
            style="transition-duration: 200ms;"
          />
        </div>
      </div>
      <div
        class="arco-avatar-info"
      >
        <div
          class="arco-avatar-name arco-avatar-name-size-ultra-small"
        >
          Username
        </div>
        <div
          class="arco-avatar-desc arco-avatar-desc-size-ultra-small"
        >
          Auxiliary information
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`avatar demo test avatar demo: custom.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="avatar-demo-box"
  >
    <div
      class="arco-avatar-wrapper small circle arco-avatar-wrapper-shape-circle"
    >
      <div
        class="arco-avatar arco-avatar-size-small small arco-avatar-shape-circle circle arco-image-avatar arco-avatar-mode-image"
      >
        <div
          class="arco-image all-border-box pc arco-avatar-img"
        >
          <div
            class="image-content image-bottom-overlap"
          >
            <div
              class="image-placeholder"
            />
          </div>
          <div
            class="image-container loading animate"
            style="transition-duration: 200ms;"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`avatar demo test avatar demo: default.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="avatar-demo-box"
  >
    <div
      class="arco-avatar-wrapper small circle arco-avatar-wrapper-shape-circle"
    >
      <div
        class="arco-avatar arco-avatar-size-small small arco-avatar-shape-circle circle arco-avatar-default-overlap default-overlap"
      >
        <svg
          class="arco-icon arco-icon-user-fill arco-avatar-default arco-avatar-default-icon-size-small"
          fill="currentColor"
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.5 10.833c2.301 0 5 1.786 5 5v2.5c0 .46-.373.834-.833.834H3.333a.833.833 0 01-.833-.834v-2.5c0-3.211 2.699-5 5-5h5zM10 1.25a4.167 4.167 0 110 8.333 4.167 4.167 0 010-8.333z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      class="arco-avatar-wrapper avatar-left-margin small square arco-avatar-wrapper-shape-square"
    >
      <div
        class="arco-avatar arco-avatar-size-small small arco-avatar-shape-square square arco-avatar-default-overlap default-overlap"
      >
        <svg
          class="arco-icon arco-icon-user-fill arco-avatar-default arco-avatar-default-icon-size-small"
          fill="currentColor"
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.5 10.833c2.301 0 5 1.786 5 5v2.5c0 .46-.373.834-.833.834H3.333a.833.833 0 01-.833-.834v-2.5c0-3.211 2.699-5 5-5h5zM10 1.25a4.167 4.167 0 110 8.333 4.167 4.167 0 010-8.333z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`avatar demo test avatar demo: icon-avatar.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="avatar-demo-box custom-icon"
  >
    <div
      class="arco-avatar-wrapper small circle arco-avatar-wrapper-shape-circle"
    >
      <div
        class="arco-avatar arco-avatar-size-small small arco-avatar-shape-circle circle"
      >
        <svg
          class="arco-icon arco-icon-user-fill custom-avatar-icon"
          fill="currentColor"
          height="1em"
          viewBox="0 0 20 20"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12.5 10.833c2.301 0 5 1.786 5 5v2.5c0 .46-.373.834-.833.834H3.333a.833.833 0 01-.833-.834v-2.5c0-3.211 2.699-5 5-5h5zM10 1.25a4.167 4.167 0 110 8.333 4.167 4.167 0 010-8.333z"
            fill="currentColor"
          />
        </svg>
        <div
          class="arco-avatar-decoration"
        >
          <img
            class="bottom-edit-icon"
            src="https://sf1-cdn-tos.toutiaostatic.com/obj/arco-mobile/_static_/avatar-edit.png"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`avatar demo test avatar demo: size.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="avatar-demo-box"
  >
    <div
      class="avatar-size-demo-box"
    >
      <div
        class="arco-avatar-wrapper large circle arco-avatar-wrapper-shape-circle"
      >
        <div
          class="arco-avatar arco-avatar-size-large large arco-avatar-shape-circle circle arco-image-avatar arco-avatar-mode-image"
        >
          <div
            class="arco-image all-border-box pc arco-avatar-img"
          >
            <div
              class="image-content image-bottom-overlap"
            >
              <div
                class="image-placeholder"
              />
            </div>
            <div
              class="image-container loading animate"
              style="transition-duration: 200ms;"
            />
          </div>
        </div>
      </div>
      <span
        class="size-text"
        style="margin-top: 4px;"
      >
        large
      </span>
    </div>
    <div
      class="avatar-size-demo-box avatar-left-margin"
    >
      <div
        class="arco-avatar-wrapper medium circle arco-avatar-wrapper-shape-circle"
      >
        <div
          class="arco-avatar arco-avatar-size-medium medium arco-avatar-shape-circle circle arco-image-avatar arco-avatar-mode-image"
        >
          <div
            class="arco-image all-border-box pc arco-avatar-img"
          >
            <div
              class="image-content image-bottom-overlap"
            >
              <div
                class="image-placeholder"
              />
            </div>
            <div
              class="image-container loading animate"
              style="transition-duration: 200ms;"
            />
          </div>
        </div>
      </div>
      <span
        class="size-text"
        style="margin-top: 8px;"
      >
        medium
      </span>
    </div>
    <div
      class="avatar-size-demo-box avatar-left-margin"
    >
      <div
        class="arco-avatar-wrapper small circle arco-avatar-wrapper-shape-circle"
      >
        <div
          class="arco-avatar arco-avatar-size-small small arco-avatar-shape-circle circle arco-image-avatar arco-avatar-mode-image"
        >
          <div
            class="arco-image all-border-box pc arco-avatar-img"
          >
            <div
              class="image-content image-bottom-overlap"
            >
              <div
                class="image-placeholder"
              />
            </div>
            <div
              class="image-container loading animate"
              style="transition-duration: 200ms;"
            />
          </div>
        </div>
      </div>
      <span
        class="size-text"
        style="margin-top: 12px;"
      >
        small
      </span>
    </div>
    <div
      class="avatar-size-demo-box avatar-left-margin"
    >
      <div
        class="arco-avatar-wrapper smaller circle arco-avatar-wrapper-shape-circle"
      >
        <div
          class="arco-avatar arco-avatar-size-smaller smaller arco-avatar-shape-circle circle arco-image-avatar arco-avatar-mode-image"
        >
          <div
            class="arco-image all-border-box pc arco-avatar-img"
          >
            <div
              class="image-content image-bottom-overlap"
            >
              <div
                class="image-placeholder"
              />
            </div>
            <div
              class="image-container loading animate"
              style="transition-duration: 200ms;"
            />
          </div>
        </div>
      </div>
      <span
        class="size-text"
        style="margin-top: 16px;"
      >
        smaller
      </span>
    </div>
    <div
      class="avatar-size-demo-box avatar-left-margin"
    >
      <div
        class="arco-avatar-wrapper ultra-small circle arco-avatar-wrapper-shape-circle"
      >
        <div
          class="arco-avatar arco-avatar-size-ultra-small ultra-small arco-avatar-shape-circle circle arco-image-avatar arco-avatar-mode-image"
        >
          <div
            class="arco-image all-border-box pc arco-avatar-img"
          >
            <div
              class="image-content image-bottom-overlap"
            >
              <div
                class="image-placeholder"
              />
            </div>
            <div
              class="image-container loading animate"
              style="transition-duration: 200ms;"
            />
          </div>
        </div>
      </div>
      <span
        class="size-text"
        style="margin-top: 20px;"
      >
        ultra-small
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`avatar demo test avatar demo: text-avatar.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="avatar-demo-box"
  >
    <div
      class="arco-avatar-wrapper small circle arco-avatar-wrapper-shape-circle"
    >
      <div
        class="arco-avatar arco-avatar-size-small small arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
      >
        <span
          class="arco-avatar-text arco-avatar-text-size-small"
          style="transform: scale(1);"
        >
          Nietzsche
        </span>
      </div>
    </div>
    <div
      class="arco-avatar-wrapper avatar-left-margin small circle arco-avatar-wrapper-shape-circle"
    >
      <div
        class="arco-avatar arco-avatar-size-small small arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
        style="background-color: rgb(255, 125, 0);"
      >
        <span
          class="arco-avatar-text arco-avatar-text-size-small"
          style="transform: scale(1);"
        >
          Z
        </span>
      </div>
    </div>
    <div
      class="arco-avatar-wrapper avatar-left-margin small square arco-avatar-wrapper-shape-square"
    >
      <div
        class="arco-avatar arco-avatar-size-small small arco-avatar-shape-square square arco-text-avatar arco-avatar-mode-text"
        style="background-color: rgb(201, 205, 212);"
      >
        <span
          class="arco-avatar-text arco-avatar-text-size-small"
          style="transform: scale(1);"
        >
          T
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`avatar demo test avatar demo: text-avatar-size.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="avatar-demo-box"
  >
    <div
      class="arco-avatar-wrapper large circle arco-avatar-wrapper-shape-circle"
    >
      <div
        class="arco-avatar arco-avatar-size-large large arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
      >
        <span
          class="arco-avatar-text arco-avatar-text-size-large"
          style="transform: scale(1);"
        >
          Nietzsche
        </span>
      </div>
    </div>
    <div
      class="arco-avatar-wrapper avatar-left-margin medium circle arco-avatar-wrapper-shape-circle"
    >
      <div
        class="arco-avatar arco-avatar-size-medium medium arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
      >
        <span
          class="arco-avatar-text arco-avatar-text-size-medium"
          style="transform: scale(1);"
        >
          Nietzsche
        </span>
      </div>
    </div>
    <div
      class="arco-avatar-wrapper avatar-left-margin small circle arco-avatar-wrapper-shape-circle"
    >
      <div
        class="arco-avatar arco-avatar-size-small small arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
      >
        <span
          class="arco-avatar-text arco-avatar-text-size-small"
          style="transform: scale(1);"
        >
          Nietzsche
        </span>
      </div>
    </div>
    <div
      class="arco-avatar-wrapper avatar-left-margin smaller circle arco-avatar-wrapper-shape-circle"
    >
      <div
        class="arco-avatar arco-avatar-size-smaller smaller arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
      >
        <span
          class="arco-avatar-text arco-avatar-text-size-smaller"
          style="transform: scale(1);"
        >
          Nietzsche
        </span>
      </div>
    </div>
    <div
      class="arco-avatar-wrapper avatar-left-margin ultra-small circle arco-avatar-wrapper-shape-circle"
    >
      <div
        class="arco-avatar arco-avatar-size-ultra-small ultra-small arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
      >
        <span
          class="arco-avatar-text arco-avatar-text-size-ultra-small"
          style="transform: scale(1);"
        >
          Nietzsche
        </span>
      </div>
    </div>
  </div>
  <div
    class="avatar-group-demo"
  >
    <div
      class="demo-sub-title"
    >
      Stacking in middle size
    </div>
    <div
      class="arco-avatar-group arco-avatar-group-size-small group-small"
    >
      <div
        class="arco-avatar-wrapper small circle arco-avatar-wrapper-shape-circle"
        style="z-index: 5;"
      >
        <div
          class="arco-avatar arco-avatar-size-small small arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
          style="background-color: rgb(123, 198, 22);"
        >
          <span
            class="arco-avatar-text arco-avatar-text-size-small"
            style="transform: scale(1);"
          >
            Nietzsche
          </span>
        </div>
      </div>
      <div
        class="arco-avatar-wrapper small circle arco-avatar-wrapper-shape-circle"
        style="z-index: 4;"
      >
        <div
          class="arco-avatar arco-avatar-size-small small arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
          style="background-color: rgb(20, 201, 201);"
        >
          <span
            class="arco-avatar-text arco-avatar-text-size-small"
            style="transform: scale(1);"
          >
            M
          </span>
        </div>
      </div>
      <div
        class="arco-avatar-wrapper small circle arco-avatar-wrapper-shape-circle"
        style="z-index: 3;"
      >
        <div
          class="arco-avatar arco-avatar-size-small small arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
          style="background-color: rgb(22, 140, 255);"
        >
          <span
            class="arco-avatar-text arco-avatar-text-size-small"
            style="transform: scale(1);"
          >
            X
          </span>
        </div>
      </div>
      <div
        class="arco-avatar-wrapper small circle arco-avatar-wrapper-shape-circle"
        style="z-index: 2;"
      >
        <div
          class="arco-avatar arco-avatar-size-small small arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
          style="background-color: rgb(255, 125, 0);"
        >
          <span
            class="arco-avatar-text arco-avatar-text-size-small"
            style="transform: scale(1);"
          >
            Z
          </span>
        </div>
      </div>
      <div
        class="arco-avatar-wrapper small circle arco-avatar-wrapper-shape-circle"
        style="z-index: 1;"
      >
        <div
          class="arco-avatar arco-avatar-size-small small arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
          style="background-color: rgb(255, 199, 46);"
        >
          <span
            class="arco-avatar-text arco-avatar-text-size-small"
            style="transform: scale(1);"
          >
            JD
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="avatar-group-demo"
  >
    <div
      class="demo-sub-title"
    >
      Stacking in small size
    </div>
    <div
      class="arco-avatar-group arco-avatar-group-size-ultra-small group-ultra-small"
    >
      <div
        class="arco-avatar-wrapper ultra-small circle arco-avatar-wrapper-shape-circle"
        style="z-index: 1;"
      >
        <div
          class="arco-avatar arco-avatar-size-ultra-small ultra-small arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
          style="background-color: rgb(123, 198, 22);"
        >
          <span
            class="arco-avatar-text arco-avatar-text-size-ultra-small"
            style="transform: scale(1);"
          >
            T
          </span>
        </div>
      </div>
      <div
        class="arco-avatar-wrapper ultra-small circle arco-avatar-wrapper-shape-circle"
        style="z-index: 2;"
      >
        <div
          class="arco-avatar arco-avatar-size-ultra-small ultra-small arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
          style="background-color: rgb(20, 201, 201);"
        >
          <span
            class="arco-avatar-text arco-avatar-text-size-ultra-small"
            style="transform: scale(1);"
          >
            M
          </span>
        </div>
      </div>
      <div
        class="arco-avatar-wrapper ultra-small circle arco-avatar-wrapper-shape-circle"
        style="z-index: 3;"
      >
        <div
          class="arco-avatar arco-avatar-size-ultra-small ultra-small arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
          style="background-color: rgb(22, 140, 255);"
        >
          <span
            class="arco-avatar-text arco-avatar-text-size-ultra-small"
            style="transform: scale(1);"
          >
            X
          </span>
        </div>
      </div>
      <div
        class="arco-avatar-wrapper ultra-small circle arco-avatar-wrapper-shape-circle"
        style="z-index: 4;"
      >
        <div
          class="arco-avatar arco-avatar-size-ultra-small ultra-small arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
          style="background-color: rgb(255, 125, 0);"
        >
          <span
            class="arco-avatar-text arco-avatar-text-size-ultra-small"
            style="transform: scale(1);"
          >
            Z
          </span>
        </div>
      </div>
      <div
        class="arco-avatar-wrapper ultra-small circle arco-avatar-wrapper-shape-circle"
        style="z-index: 5;"
      >
        <div
          class="arco-avatar arco-avatar-size-ultra-small ultra-small arco-avatar-shape-circle circle arco-text-avatar arco-avatar-mode-text"
          style="background-color: rgb(255, 199, 46);"
        >
          <span
            class="arco-avatar-text arco-avatar-text-size-ultra-small"
            style="transform: scale(1);"
          >
            JD
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
